### PR TITLE
Update kseq++.hpp

### PR DESCRIPTION
**Pull request by rob-p**
Add optimization originally due to @kloetzl in the original kseq.h.

### CODE REVIEW

**Comment by kloetzl** [comment]
Thanks for taking the initiative in porting my optimization. Just two comments. 1) It looks like the indentation got messed up. 2) You may want to compare against `nullptr` to keep it modern.

File: include/kseq++/kseq++.hpp [hunk 1]
```diff
@@ -678,9 +678,12 @@ namespace klibpp {
...
             --this->begin;
             if ( delimiter == KStream::SEP_LINE ) {
-              for ( i = this->begin; i < this->end; ++i ) {
-                if ( this->buf[ i ] == '\n' ) break;
-              }
+              // Incorporate optimization from new seqtk (see : https://github.com/lh3/seqtk/pull/123)
+              // Fabian commmented on this here (https://twitter.com/kloetzl/status/1661679452479266818)
+              // and suggested that std::find() may be more idiomatic.  However, I'm a bit concerned
+              // that may be non-trivially slower than memchr (https://gms.tf/stdfind-and-memchr-optimizations.html).
+              unsigned char *sep = std:::memchr(ks->buf + ks->begin, '\n', ks->end - ks->begin);
```
Comment (kloetzl) [inline include/kseq++/kseq++.hpp:685]: One too many `:`.

**Review by kloetzl — APPROVED** [review]

**Comment by cartoonist** [comment]
Thanks for creating the PR! I just (manually) merged it to the develop branch.

**Comment by cartoonist** [comment]
Now, it is included in release v1.1.1.

**Pull request closed without merging**